source/deploy/postgres-migration.rst

```diff
@@ -6,32 +6,167 @@ Migration guidelines from MySQL to PostgreSQL
 
 From Mattermost v8.0, PostgreSQL is our database of choice for Mattermost to enhance the platform’s performance and capabilities. Recognizing the importance of supporting the community members who are interested in migrating from a MySQL database, we have taken proactive measures to provide guidance and best practices. 
 
-To streamline the migration process and alleviate any potential challenges, we have prepared a comprehensive set of guidelines to facilitate a smooth transition. Additionally, we want to offer recommendations for various tools that have proven to be highly effective in simplifying your migration efforts.
+To streamline the migration process and alleviate potential challenges, we have prepared a comprehensive set of guidelines and a `migration-assist  tool <#migration-assist-tool>`__ to facilitate a smooth transition. Additionally, we offer `tool recommendations <#manual-migration-tool-recommendations>`__ and `manual migration guidance <#system-requirements-and-configurations-for-manual-migrations>`__ if you prefer to migrate manually.
 
-.. note::
+migration-assist tool
+---------------------
+
+Migrating databases can be a daunting task, and it can be easy to overlook or misinterpret some of the required steps if you haven't performed a migration before. 
+
+Our ``migration-assist`` tool provides an efficient, error-free migration experience that automates the tasks to be executed. The tool offers 3 core utility commands:
+
+1. ``migration-assist mysql``
+
+   Checks the MySQL database schema to ensure readiness for migration, and offers fixes for common issues.
+
+2. ``migration-assist postgres``
+
+   Creates the PostgreSQL database schema, and prepares it for Mattermost deployment by downloading the necessary migrations and applying them.
+
+3. ``migration-assist pgloader``
+
+   Generates a pgloader configuration from DSN values, ensuring accurate data transfer.
+
+Install
+~~~~~~~
+
+The ``migration-assist`` tool can be downloaded and compiled with the `Go <https://go.dev/>`__ toolchain. The tool requires at least ``v1.22`` of the Go compiler.
+
+Use ``go install`` to install the tool:
+
+.. code-block:: shell
+
+   go install github.com/mattermost/migration-assist/cmd/migration-assist@latest
+
+Usage
+~~~~~~
+
+.. important::
+
+   Please make sure you have the necessary environment to perform the migration. Ensure that the MySQL and PostgreSQL databases are running and accessible. To setup a PostgreSQL instance, see the :doc:`database </install/prepare-mattermost-database>` documentation for details.
+
+Step 1 - Check the MySQL database schema
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Run the following command to check the MySQL database schema:
+
+.. code-block:: shell
+
+   migration-assist mysql "<MYSQL_DSN>"
```

Comment:
I think it might be helpful to give our readers an example of a `MYSQL_DSN`. Same for other places on the page.

Reply:
@isacikgoz - Can you draft these details, please?